explorer/interpreter/action.h

```diff
@@ -50,7 +50,6 @@ class RuntimeScope {
 
   // Allocates storage for `value_node` in `heap`, and initializes it with
   // `value`.
-  // TODO: Update existing callers to use Bind instead, where appropriate.
```

Comment:
I think this TODO is backwards: we should stop using `Bind` entirely and remove it, because what it's doing (go looking for any object that has the given value and bind `value_node` to that object) doesn't make sense.

We should bind `ValueNodeView`s to `AllocationId`s, not to `Value`s.

Reply:
In my mind `Bind` would be needed, independently of this current design issue (though its signature would likely change), to bind a `ValueNodeView` to an `AllocationId` without adding it to `allocations_` (which are then destroyed when backtracking).
The sibling `Initialize` method on the other hand is guaranteed to create an AllocationId, and to add it to `allocations_` for later destruction+dealloc.

I think this is somewhat orthogonal to the other work on virtual destructors and the alloc/dealloc problem.

Does that makes sense though, or do you think the impact of that other change makes this one void?

Reply:
I think you're right, but I don't think we should move more callers over to using `Bind` until it has the right semantics. Using `Initialize` is wrong, but mostly not observably so. But using `Bind` instead will be observably wrong in surprising ways. For example, given:

```
fn F(n: i32, p: i32*) -> i32 {
  *p = 2;
  return n;
}
fn Main() -> i32 {
  var k: i32 = 1;
  var j: i32 = k;
  return F(j, &k);
}
```

... today, `Main` returns `1`, but with this change I believe it'll return `2`, because we bind main's `k` object to `F`'s `n` parameter (because that happens to be the first live allocation whose value is the `2` `IntValue` passed to `F`, as `k` and `j` hold the same `IntValue`), and the assignment through `*p` thereby changes the observed value of `n`.